src/events/logs.ts

```diff
@@ -0,0 +1,385 @@
+            { name: "Message ID", value: before.id, inline: true },
+            { name: "Channel", value: before.channel.toString(), inline: false }
+        );
+
+    // Add this if the content was updated,
+    // and if the old message actually had text
+    if (
+        oldContent !== newContent &&
+        oldContent
+    ) {
+        embed.addFields(
+            { name: "Old content", value: oldContent, inline: false },
+            { name: "New content", value: newContent ? newContent : "_ _", inline: false }
+        );
+    }
+
+    // Add this if the attachments were updated
+    if (diffAttachments) {
+        // Get the URLs of the old attachmetns
+        let oldUrls: string = "";
+
+        for (const [, attachment] of oldAttachments) {
+            oldUrls += `${attachment.url}\n`;
+        }
+
+        // Get the URLS of the new attachments
+        let newUrls: string = "";
+
+        for (const [, attachment] of newAttachments) {
+            newUrls += `${attachment.url}\n`;
+        }
+
+        embed.addFields(
+            { name: "Old attachments", value: oldUrls ? oldUrls : "_ _", inline: false },
+            { name: "New attachments", value: newUrls ? newUrls : "_ _", inline: false }
+        );
+    }
+
+    return embed;
+};
+
+/**
+ * Generates an embed for whenever a message is deleted.
+ * 
+ * @param message the deleted message
+ * @returns en embed for logging,
+ * or null if the message had neithe text nor attachments
+ */
+export const genMessageDeleteEmbed = (message: Message): EmbedBuilder | null => {
+    // The components to log
+    const content: string = message.content;
+    const attachments: Collection<string, Attachment> = message.attachments;
+    const author: User = message.author;
+
+    // No embed generated if the message had no content nor attachments
+    if (!content && !attachments.size) {
+        return null;
+    }
+
+    const embed: EmbedBuilder = new EmbedBuilder()
+        .setTitle(author.username)
+        .setColor(ORANGE)
+        .setAuthor({
+            name: "Message deleted",
+            iconURL: message.guild.iconURL()
+        })
+        .setTimestamp(message.createdAt)
+        .addFields(
+            { name: "Member ID", value: author.id, inline: true },
+            { name: "Message ID", value: message.id, inline: true },
+            { name: "Channel", value: message.channel.toString(), inline: false },
+        );
+
+    // Add if the message has text
+    if (content) {
+        embed.addFields(
+            { name: "Content", value: content, inline: false }
+        );
+    }
+
+    // Add if the message has attachments
+    if (attachments.size) {
+        let urls: string = "";
+
+        for (const [, attachment] of attachments) {
+            urls += `${attachment.url}\n`;
+        }
+
+        embed.addFields(
+            { name: "Attachments", value: urls, inline: false }
+        );
+    }
+
+    return embed;
+};
```

Comment:
These are the embeds used for message changes
They check the message's content and attachments